Fix torch.tensor call with tuple in Dynamo

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114383

Partially fixes https://github.com/pytorch/pytorch/issues/114231

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

ezyang committed:
Fix torch.tensor call with tuple in Dynamo

Partially fixes https://github.com/pytorch/pytorch/issues/114231

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

ezyang committed:
Update on "Fix torch.tensor call with tuple in Dynamo"


Partially fixes https://github.com/pytorch/pytorch/issues/114231

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot rebase

voznesenskym commented on torch/_dynamo/variables/torch.py:
```diff
@@ -575,7 +575,7 @@ def check_any_unspec(x):
                     if isinstance(x, (TensorVariable, SymNodeVariable)):
                         return True
-                    elif isinstance(x, ListVariable):
+                    elif isinstance(x, (ListVariable, TupleVariable)):
```
the dynamo pattern here is unpack_var_sequence check and access, should work for all collections.

ezyang committed:
Update on "Fix torch.tensor call with tuple in Dynamo"


Partially fixes https://github.com/pytorch/pytorch/issues/114231

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

voznesenskym commented on torch/_dynamo/variables/torch.py:
```diff
@@ -575,7 +575,7 @@ def check_any_unspec(x):
-                    elif isinstance(x, ListVariable):
+                    elif isinstance(x, (ListVariable, TupleVariable)):
                         return any(check_any_unspec(y) for y in x.items)
                     # TODO: there maybe other recursive structures you need to
                     # check
```
reeeeee this whole func needs a good hard revisit

ezyang commented:
looks like someone merged something equivalent
